code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,272 @@
+		prev = current
+		//i tried using switch, but byond is really fucky and it didnt work as intended. Im sorry
+		if(i == 0)
+			current = new type(drop_location(),spawn_more = FALSE)
+			current.icon_state = "armsy_mid"
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		else if(i > 0 && i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		else
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/can_be_pulled()
+	return FALSE
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+	. = ..()
+	gib_trail()
+	if(back && back.loc != oldloc)
+		back.Move(oldloc)
+	if(front && loc != front.oldloc)
+		forceMove(front.oldloc) // self fixing properties
+	oldloc = loc
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/gib_trail()
+	if(front) // head makes gibs
+		return
+	var/chosen_decal = pick(typesof(/obj/effect/decal/cleanable/blood/gibs))
+	var/obj/effect/decal/cleanable/blood/gibs/decal = new chosen_decal(drop_location())
+	decal.setDir(dir)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Destroy()
+	if(front)
+		front.icon_state = "armsy_end"
+		front.icon_living = "armsy_end"
+	if(back)
+		qdel(back) // chain destruction baby
+	return ..()
+
+
+/mob/living/simple_animal/hostile/eldritch/armsy/proc/heal()
+	if(health == maxHealth)
+		if(back)
+			back.heal()
+			return
+		else
+			current_stacks++
+			if(current_stacks >= stacks_to_grow)
+				var/mob/living/simple_animal/hostile/eldritch/armsy/prev = new type(drop_location(),spawn_more = FALSE)
+				icon_state = "armsy_mid"
+				icon_living =  "armsy_mid"
+				back = prev
+				prev.icon_state = "armsy_end"
+				prev.icon_living = "armsy_end"
+				prev.front = src
+				prev.AIStatus = AI_OFF
+
+				current_stacks = 0
+
+	adjustBruteLoss(-maxHealth * 0.5, FALSE)
+	adjustFireLoss(-maxHealth * 0.5 ,FALSE)
+	adjustToxLoss(-maxHealth * 0.5, FALSE)
+	adjustOxyLoss(-maxHealth * 0.5)
+
+/mob/living/simple_animal/hostile/eldritch/armsy/AttackingTarget()
+	if(istype(target,/obj/item/bodypart/r_arm) || istype(target,/obj/item/bodypart/l_arm))
+		qdel(target)
+		heal()
+		return
+	if(back)
+		back.target = target
+		back.AttackingTarget()
+	if(!Adjacent(target))
+		return
+	do_attack_animation(target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(HAS_TRAIT(C, TRAIT_NODISMEMBER))
+			return
+		var/list/parts = list()
+		for(var/X in C.bodyparts)
+			var/obj/item/bodypart/BP = X
+			if(BP.body_part != HEAD && BP.body_part != CHEST && BP.body_part != LEG_LEFT && BP.body_part != LEG_RIGHT)
+				if(BP.dismemberable)
+					parts += BP
+		if(length(parts) && prob(10))
+			var/obj/item/bodypart/BP = pick(parts)
+			BP.dismember()
```

Comment:
Avoid one or two uppercase letters as variable names.
Use descriptive ones.

Reply:
![image](https://user-images.githubusercontent.com/29008542/82951132-a024a400-9f74-11ea-9cb8-223f0559d7d2.png)
call me a moron but i feel like the first code block is much more readable

Reply:
eh whatever i will replace it.